rule.go

```diff
@@ -84,7 +84,10 @@ func (cc *Conn) GetRule(t *Table, c *Chain) ([]*Rule, error) {
 		if err != nil {
 			return nil, err
 		}
-		rules = append(rules, r)
+
+		if r.Table.Name == t.Name && r.Table.Family == t.Family && r.Chain.Name == c.Name {
```

Comment:
If I read https://elixir.bootlin.com/linux/v5.15.2/source/net/netfilter/nf_tables_api.c#L3136 correctly, I think setting the NLM_F_ECHO flag (which we do in line 68) results in dumping all rules regardless of the arguments.

So, the better fix might be to not set NLM_F_ECHO, so that the kernel only returns the data we are requesting?

Reply:
Yes, your analysis is correct. At first, I only considered the frame selection of data at this layer, but ignored the restrictions when the c language api was first adjusted. Testing is also possible.